core/src/main/scala/org/apache/spark/scheduler/SparkListener.scala

```diff
@@ -249,6 +250,8 @@ trait SparkListener {
    * Called when other events like SQL-specific events are posted.
    */
   def onOtherEvent(event: SparkListenerEvent) { }
+
+  // WHENEVER WE ADD A METHOD HERE, PLEASE ALSO UPDATE SparkFirehoseListener.
```

Comment:
We can have both this abstract class and also an interface? That way the SparkFirehoseListener can just implement the interface instead of extending the abstract class. Not sure if that is a better situation or not - it does make it more flexible though.